### PR TITLE
In check

### PR DESCRIPTION
**Pull request by snsmith92**

### CODE REVIEW

File: app/models/game.rb [hunk 1]
```diff
@@ -38,6 +37,53 @@ def populate_game!
...
+    end 
+  end 
+
+  private
+
+  def piece_params
+    params.require(:piece).permit(:type, :position_x, :position_y, :game_id, :color, :captured, :image)
+  end
+
+  def black_in_check
```
[the remark this answers is not in the document]
Comment (snsmith92) [reply app/models/game.rb]: Thanks @gvin10 !

File: app/models/game.rb [hunk 2]
```diff
@@ -38,8 +37,55 @@ def populate_game!
...
+  def black_in_check?
+    black_king = self.pieces.find_by(type: 'King', color: 'black')
+    position_x = black_king.position_x
+    position_y = black_king.position_y
+
+    self.pieces.each do |piece|
+      if piece.valid_move?(position_x, position_y) && piece.color == 'white'
+        return true
+      else
+        return false
```
Comment (ilyakrasnov) [inline app/models/game.rb]: returning false inside `each` block will mean that if the first of the block does not threaten the king, the whole method will return false.. you want to return true inside the block if you meet the first piece threatening the king, and outside the block return false (meaning that no piece from `self.pieces` was a threat)

File: app/models/game.rb [hunk 3]
```diff
@@ -38,8 +37,55 @@ def populate_game!
...
+    end 
+  end 
+
+  def black_in_check?
+    black_king = self.pieces.find_by(type: 'King', color: 'black')
+    position_x = black_king.position_x
+    position_y = black_king.position_y
+
+    self.pieces.each do |piece|
+      if piece.valid_move?(position_x, position_y) && piece.color == 'white'
```
Comment (ilyakrasnov) [inline app/models/game.rb]: instead if/else you could just say:
`return true if piece.valid_move?(position_x, position_y) && piece.color == 'white'`

Comment (ilyakrasnov) [reply app/models/game.rb]: we could also have sth like `white_pieces = ...` and iterate over them, saving the `&& piece.color`

File: app/models/game.rb [hunk 4]
```diff
@@ -38,8 +37,55 @@ def populate_game!
     end
   end
 
+  def in_check?
+    kings = self.pieces.find_by(type: 'King')
+
```
Comment (ilyakrasnov) [inline app/models/game.rb]: if you use the loop to iterate over the kings, you will either way execute both methods..


the whole method could be only one line:

`black_in_check? || white_in_check?` 😉

**Pull request merged**